src/applications/appeals/995/utils/evidence.js

```diff
@@ -95,22 +95,30 @@ export const removeNonSelectedIssuesFromEvidence = data => {
  * @param {Object} router - React router
  */
 export const onFormLoaded = props => {
-  const { formData, returnUrl, router } = props;
+  let { returnUrl } = props;
+  const { formData, router } = props;
   const { locations = [] } = formData;
-  if (showScNewForm(formData) && locations.length) {
-    formData.locations = locations.map(location => {
-      if (!location.treatmentDate) {
-        const from = location.evidenceDates?.from || '';
-        const treatmentDate = (from || '').substring(0, from.lastIndexOf('-'));
-        const noDate = !treatmentDate;
-        return {
-          ...location,
-          treatmentDate,
-          noDate,
-        };
-      }
-      return location;
-    });
+  if (showScNewForm(formData)) {
+    returnUrl = checkRedirect(formData, returnUrl);
+
+    if (locations.length) {
+      formData.locations = locations.map(location => {
+        if (!location.treatmentDate) {
+          const from = location.evidenceDates?.from || '';
+          const treatmentDate = (from || '').substring(
+            0,
+            from.lastIndexOf('-'),
+          );
+          const noDate = !treatmentDate;
```

Comment:
`treatmentDate` is a string. Ew...

Reply:
Yeah, I have my JS blinders on most of the time... fixed!

Reply:
Eh, this is fine: `false == '' == [] == 0 // true`